### DISCUSSION TITLE
How can I set install path for a particular feature in Custom UI dialog and make managed project to take that while installing

### DISCUSSION THREAD

**Discussion by abskulkarni** [oleg-shilo/wixsharp#1412] [Q&A]
Hi
I am using WixSharp to build msi for our product.
I have a custom UI dialog where I am giving facility to user to change installation directory for a feature. (I am having multiple features those may have different installation directories/paths)
How can I make managed project to take this user defined path while installing that feature?

**Comment by oleg-shilo** [comment]
see my answer here: https://github.com/oleg-shilo/wixsharp/issues/1413

**Reply by abskulkarni** [reply]
Hi
Many thanks for your reply. Actually I did manage to get features dialog as per my requirement. Please see attached image having installation screen & related Wix# code.

I might have put my requirement in wrong words due to which you said it is not possible. But please consider attached installation screen is my requirement.

You can see that I have instantiated Feature class object with ConfigurableDir parameter. ("FEATURE1DIR" & "FEATURE2DIR").
It tells me that it will show "Browse" button if ConfiurableDir parameter is non-empty.

However, my question or blocking point is, **I don't get Browse button beside every feature**. Browse button will be used by user to set installation directory for that feature.

Can you please help me in getting Browse button enabled there?

![WixSharpFeatureWithConfigurableDirectory](https://github.com/oleg-shilo/wixsharp/assets/23351437/8018c31b-f7c9-486b-9aa7-6042f96c8707)

**Comment by oleg-shilo** [comment]
I am not sure I follow.
You are still using the same problem description as in you original post:

> However, my question or blocking point is, I don't get Browse button beside every feature. Browse button will be used by user to set installation directory for that feature.

I was trying to explain that the installation directory the user provides is used for the installation of the product. For all selected features. Not "for that feature".

I also see the first dir in your code snippet that has the id set to some path. It's not what is expected for the id. Have a look at some samples to get an idea about a typical setup definition. IE `InstallDir`, `InstallDir_AbsolutePath`, etc.
